authors/apps/notifications/backends.py

```diff
@@ -0,0 +1,199 @@
+                message = self._message_builder(
+                    sender, "left a  comment on an article you favorited,",
+                    comment.article.title
+                )
+                profile.user.email
+                self._dispatch(request, (sender,
+                               profile, message, action_link,),
+                               'FAVORITED_ARTICLE_COMMENTED')
+
+    def article_interaction_liked_or_faved(
+            self, request, article, liked_by=None, favorited_by=None):
+        """
+        notify users that created an article of
+        a new like or favorite
+
+        args:
+            - request (Request): the commenter's request object
+            - article(ArticleModel): the article object
+            - liked_by (ProfileModel): the liker's profile object, Defaults to None
+            - favorited_by (ProfileModel): the favoriter's profile object, Defaults to None  
+
+        """
+        sender = liked_by or favorited_by
+        receiver = article.author
+
+        if self._is_same_profile(sender, receiver):
+            return
+        
+        action_link = request.build_absolute_uri(
+            reverse('articles:article-update', kwargs={'slug': article.slug}))
+           
+        if liked_by is not None:
+            message = self._message_builder(sender,
+                                            "liked your article",
+                                            article.title
+                                            )
+        
+            self._dispatch(request, (sender,
+                           receiver, message, action_link,),
+                           'AUTHORED_ARTICLE_LIKED')
+       
+        if favorited_by is not None:
+            sender_profile = sender
+            message = self._message_builder(sender_profile,
+                                            "favorited your article",
+                                            article.title
+                                            )
+         
+            self._dispatch(request, (sender,
+                           receiver, message, action_link,),
+                           'AUTHORED_ARTICLE_FAVORITED')
+
+    def _is_same_profile(self, sender, receiver):
+        """
+        checks if sender profile is receiver profile
+
+        Args:
+            - sender (ProfileModel): the sender's profile
+            - receiver (ProfileModel): the receiver's profile
+
+        return True or False
+
+        """
+        return sender.user.username == receiver.user.username
+
+    def _message_builder(self, person, message, title):
+        """
+        builds a notification message string
+        Args:
+            - person(ProfileModel): person mentioned in the notification
+            - message(String): message describing changes
+            - title(String): affected article title
+
+        returns:
+            - notification_message(String)
+        """
+        return self._get_name(person)\
```

Comment:
@malfahad use format. it is much cleaner than concatenating strings using the `+` operator.